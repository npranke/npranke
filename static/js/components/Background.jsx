import React from 'react'

import Landscape from '@img/background-landscape.jpg'
import Portrait from '@img/background-portrait.jpg'

function Background() {
    return (
        <picture>
            <source srcSet={ Portrait } media="(orientation: portrait)" />
            <img
                src={ Landscape }
                className="background"
                alt="Conifer in Colorado landscape"
            />
        </picture>
    )
}

export default Background
