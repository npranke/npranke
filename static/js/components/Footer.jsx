function Footer() {
    const currentYear = new Date().getFullYear()

    return (
        <footer className="footer" role="contentinfo">
            <div className="copyright">
                &copy; 2017-{currentYear} Nicole Pranke
            </div>
        </footer>
    )
}

export default Footer
