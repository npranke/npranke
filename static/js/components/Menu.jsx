import React from 'react'

function Menu() {
    return (
        <div className="menu">
            <div className="intro">
                Hi, I&#39;m a software engineer; my name is Nicole.
            </div>
            <ul>
                <li>
                    <a href="#" target="_self" rel="noreferrer">
                        small useful things
                    </a>
                </li>
                <li>
                    <a href="#" target="_blank" rel="noreferrer noopener">
                        linkedin
                    </a>
                </li>
            </ul>
        </div>
    )
}

export default Menu
