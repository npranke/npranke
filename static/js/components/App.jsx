import React from 'react'

import Background from './Background'
import Footer from './Footer'
import Menu from './Menu'

function App() {
    return (
        <div className="app">
            <Background />
            <Menu />
            <Footer />
        </div>
    )
}

export default App
