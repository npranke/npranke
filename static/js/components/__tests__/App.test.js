import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import App from '../App'
import Background from '../Background'
import Footer from '../Footer'
import Menu from '../Menu'

Enzyme.configure({ adapter: new Adapter() })

describe('App', () => {
    test('contains background', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Background />),
        ).toBe(true)
    })

    test('contains menu', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Menu />),
        ).toBe(true)
    })

    test('contains footer', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Footer />),
        ).toBe(true)
    })
})

describe('App snapshot', () => {
    test('matches snapshot', () => {
        const app = shallow(<App />)

        expect(app).toMatchSnapshot()
    })
})
