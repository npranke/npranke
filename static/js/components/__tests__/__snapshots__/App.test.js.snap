// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot matches snapshot 1`] = `
<div
  className="app"
>
  <Background />
  <Menu />
  <Footer />
</div>
`;
